### PR TITLE
Add option with-new-pkgs to apt update

### LINKED ISSUE #4: Apt Update should expose option --with-new-pkgs
Apt update should give the option to use `--with-new-pkgs` in case of updates which cannot be made otherwise, like kernel packages update.

### PR DESCRIPTION
**Pull request by skatsaounis**

### CODE REVIEW

File: amaltheia/update.py [hunk 1]
```diff
@@ -95,10 +95,14 @@ class AptPackagesUpdater(Updater):
     }"""
 
     def update(self):
+        with_new_pkgs_flag = jinja(self.updater_args.get('withnewpkgs', False))
```
Comment (neoaggelos) [inline amaltheia/update.py]: I would suggest changing the option name to `with-new-pkgs` to stay consistent with the rest of the argument names

File: amaltheia/update.py [hunk 2]
```diff
@@ -95,10 +95,14 @@ class AptPackagesUpdater(Updater):
...
+        with_new_pkgs = '--with-new-pkgs' if with_newpkgs_flag else ''
+
         stdout, stderr = ssh_cmd(
             self.host, self.host_args,
             'sudo DEBIAN_FRONTEND=noninteractive apt-get -y -q'
-            ' -o Dpkg::Options::=--force-confold upgrade;')
+            ' -o Dpkg::Options::=--force-confold {} upgrade;'.format(
+                with_new_pkgs))
```
Comment (neoaggelos) [inline amaltheia/update.py]: If we are using `format` anyway, I would suggest changing the command line to something more readable instead:

```
'sudo {} apt-get upgrade -y -q {} {};'.format(
   'DEBIAN_FRONTEND=noninteractive',
   '-o  Dpkg::Options=--force-confold',
   with_new_pkgs)
```

File: docs/configuration.md [hunk 3]
```diff
@@ -318,6 +318,7 @@ for this action are described below:
 | `apt.hold`         | NO       | List    | `[apt, vim]`           | `apt-mark hold` a list of packages **before** updating                                                   |
 | `apt.unhold`       | NO       | List    | `[apt, vim]`           | `apt-mark unhold` a list of packages **after** updating                                                  |
 | `apt.fix-hostname` | NO       | String  | `{{ host }}.my.domain` | Jinja template for configuring the host name to use (if any override is needed, e.g. adding domain name) |
+| `apt.withnewpkgs`  | NO       | Boolean | `true`                 | Run `apt-get upgrade --with-new-pkgs`                                                                    |
```
Comment (neoaggelos) [inline docs/configuration.md]: Also rename here

File: examples/06-patchman.yaml [hunk 4]
```diff
@@ -6,13 +6,17 @@
 ## - filter       Only perform updates for machines whose names match
 ##                the filter
 ## - autoremove   Set to non-empty value to also autoremove old packages
+## - withnewpkgs  Set to non-empty value to also install new packages
 
 ## Usage 1 (to autoremove packages):
 ##  $ amaltheia -s this-file.yaml -v autoremove="1"
 
 ## Usage 2 (without autoremove)
```
Comment (neoaggelos) [inline examples/06-patchman.yaml]: Since you are doing it anyway on the other file, add `:` here as well

File: jenkins/job.sample.yaml [hunk 5]
```diff
@@ -6,13 +6,17 @@
...
 
 ## Usage 1 (to autoremove packages):
 ##  $ amaltheia -s this-file.yaml -v autoremove=true
 
-## Usage 2 (without autoremove)
+## Usage 2 (without autoremove):
 ##  $ amaltheia -s this-file.yaml -v filter="FILTER" autoremove=false
 
+## Usage 3 (to install new packages):
+##  $ amaltheia -s this-file.yaml -v filter="FILTER" withnewpkgs="1"
```
Comment (neoaggelos) [inline jenkins/job.sample.yaml]: `withnewpkgs=true` here as well

File: examples/06-patchman.yaml [hunk 6]
```diff
@@ -31,6 +35,7 @@ hosts:
     on-package-updates:
     - apt:
         autoremove: '{{ autoremove }}'
+        withnewpkgs: '{{ withnewpkgs }}'
```
Comment (neoaggelos) [inline examples/06-patchman.yaml]: also here. It is fine to keep the parameter as `withnewpkgs`

File: examples/06-patchman.yaml [hunk 7]
```diff
@@ -6,13 +6,17 @@
...
+## - withnewpkgs  Set to non-empty value to also install new packages
 
 ## Usage 1 (to autoremove packages):
 ##  $ amaltheia -s this-file.yaml -v autoremove="1"
 
 ## Usage 2 (without autoremove)
 ##  $ amaltheia -s this-file.yaml -v filter="FILTER" autoremove=false
 
+## Usage 3 (to install new packages):
+##  $ amaltheia -s this-file.yaml -v filter="FILTER" withnewpkgs="1"
```
Comment (neoaggelos) [inline examples/06-patchman.yaml]: `withnewpkgs=true`, since this is a boolean variable.

File: jenkins/job.sample.yaml [hunk 8]
```diff
@@ -31,6 +35,7 @@ hosts:
     on-package-updates:
     - apt:
         autoremove: '{{ autoremove }}'
+        withnewpkgs: '{{ withnewpkgs }}'
```
Comment (neoaggelos) [inline jenkins/job.sample.yaml]: `with-new-pkgs` here as well, but keep the parameter name

**Review by neoaggelos — CHANGES_REQUESTED** [review]
Requesting a few changes for consistency. Good work in updating the exampels as well.

File: jenkins/job.sample.yaml (hunk 8 above)
Comment (skatsaounis) [reply jenkins/job.sample.yaml]: done

File: examples/06-patchman.yaml (hunk 7 above)
Comment (skatsaounis) [reply examples/06-patchman.yaml]: done

File: examples/06-patchman.yaml (hunk 6 above)
Comment (skatsaounis) [reply examples/06-patchman.yaml]: done

File: jenkins/job.sample.yaml (hunk 5 above)
Comment (skatsaounis) [reply jenkins/job.sample.yaml]: done

File: examples/06-patchman.yaml (hunk 4 above)
Comment (skatsaounis) [reply examples/06-patchman.yaml]: done

File: docs/configuration.md (hunk 3 above)
Comment (skatsaounis) [reply docs/configuration.md]: done

File: amaltheia/update.py (hunk 2 above)
Comment (skatsaounis) [reply amaltheia/update.py]: done

File: amaltheia/update.py (hunk 1 above)
Comment (skatsaounis) [reply amaltheia/update.py]: done

**Review by neoaggelos — APPROVED** [review]

**Comment by skatsaounis** [comment]
@neoaggelos can we merge this one? Is there something you are waiting for?

**Pull request merged**